apps/ui/src/components/sidebar/views/favourites.tsx

```diff
@@ -1,76 +1,104 @@
 import { UnixNanoTimeStamp } from "@graphql-debugger/time";
 
-import { useContext } from "react";
+import { useContext, useMemo, useState } from "react";
 import { Link, useParams } from "react-router-dom";
 
+import { SearchBox } from "../../../components/search-box";
 import { ClientContext } from "../../../context/client";
 import { Delete } from "../../../icons/delete";
 import { IDS } from "../../../testing";
+import { traceNameIncludes } from "../../../utils/find-traces";
 import { isTraceError } from "../../../utils/is-trace-error";
 import { rootSpanName } from "../../../utils/root-span-name";
 
 export function Favourites() {
   const params = useParams<{ traceId: string }>();
   const { favourites, handleDeleteFavouriteTrace } = useContext(ClientContext);
+  const [searchFavourites, setSearchFavourites] = useState("");
 
-  const sortedfavourites = favourites.sort((a, b) => {
-    return new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime();
-  });
+  const sortedfavourites = useMemo(
+    () =>
+      favourites.sort((a, b) => {
```

Comment:
```suggestion
      […favourites].sort((a, b) => {
```
Do not mutate the original array